packages/ui/src/hooks/useCart.ts

```diff
@@ -0,0 +1,29 @@
+import { Cart, CartContext } from '../context/CartProvider'
+import { useContext, useSyncExternalStore } from 'react'
+
+export default function useCart<SelectorOutput>(
+  selector: (store: Cart) => SelectorOutput
+) {
+  const cart = useContext(CartContext)
+  if (!cart) {
+    throw new Error('Cart not found')
+  }
+
+  const data = useSyncExternalStore(
```

Comment:
is this really the right use of sync external store. don't we only really have internal state?